docs/Antora/modules/howto_guides/pages/howto_measure_single_value.adoc

```diff
@@ -1,33 +1,305 @@
+* `IInteger` if the type is `Int8` through `Int64` or `UInt8` through `UInt64`
+* `IComplexNumber` if the type is `ComplexFloat32` or `ComplexFloat64`
+* `IRange` if the type is `RangeInt64`
+* `IStruct` if the type is `Struct`
+* `IList` of the aforementioned types if there is exactly one dimension in the single value
+
+An example of casting the `lastValue` object into an `IFloat` object:
+
+[tabs]
+====
+Cpp::
++
+[source,cpp]
+----
+// Cast to IFloat
+auto number = lastValue.asPtr<IFloat>();
+----
+Python::
++
+[source,python]
+----
+
+----
+====
+
+In the above example, the Data Descriptor of the Data Packet or its Signal on which `getLastValue` (`last_value` in Python) is called might be:
+
+[tabs]
+====
+Cpp::
++
+[source,cpp]
+----
+// The Data Descriptor for Float64
+auto descriptor = DataDescriptorBuilder().setSampleType(SampleType::Float64).build();
+----
+Python::
++
+[source,python]
+----
+
+----
+====
+
+[#last_value_range]
+== Getting the last value with a Range
+
+The following example demonstrates casting the `lastValue` object into an `IRange`. Something usefeul we can then do is call the methods of the underlying type and access its members, such as the `getLowValue` (`low_value` in Python) and the `getHighValue` (`high_value` in Python), in this case provided by the `IRange` interface.
+
+[tabs]
+====
+Cpp::
++
+[source,cpp]
+----
+// Cast to IRange
+auto range = lastValue.asPtr<IRange>();
+// Extract values
+auto low = range.getLowValue();
+auto high = range.getHighValue();
+----
+Python::
++
+[source,python]
+----
+
+----
+====
+
+In the above example, the Data Descriptor of the Data Packet or its Signal on which `getLastValue` (`last_value` in Python) is called is similar to the the one from the previous example (the Data Descriptor with Sample Type `Float64`), except the Sample Type is `RangeInt64`.
+
+[#last_value_complex_number]
+== Getting the last value with a Complex Number
+
+In the following example:
+
+* Last value is extracted via a `getLastValue` (`last_value` in Python) call
+* `lastValue` object is cast into a `IComplexNumber` type (not needed in Python)
+* Methods of the underlying type such as the `getReal` (`real` in Python) and the `getImaginary` (`imaginary` in Python) are called, in this case provided by the `IComplexNumber` interface
+
+[tabs]
+====
+Cpp::
++
+[source,cpp]
+----
+// Get last value of a Signal
+auto lastValue = signal.getLastValue();
+// Cast to IComplexNumber
+auto complex = lastValue.asPtr<IComplexNumber>();
+// Extract values
+auto real = complex.getReal();
+auto imaginary = complex.getImaginary();
+----
+Python::
++
+[source,python]
+----
+
+====
+
+A Data Descriptor for this Complex Number example is again very similar to the ones from previous examples, except the Sample Type might be `ComplexFloat32` or `ComplexFloat64` or  instead of `Float64` or `RangeInt64`.
+
+[#last_value_struct]
+== Getting the last value with a Struct
+
+A Data Descriptor for a Struct might be:
```

Comment:
We should simply describe the descriptor and how the signal in question is structured. No need to provide the builder code sample.